lib/dialects/abstract/connection-manager.js

```diff
@@ -295,6 +295,11 @@ class ConnectionManager {
   releaseConnection(connection) {
     return this.pool.release(connection).tap(() => {
       debug('connection released');
+    }).catch((e) => {
+      // ISSUE(7625) No-op if generic-pool is unable to release the connection, it may have already been released.
+      if (!/Resource not currently part of this pool/.exec(e.message)) {
```

Comment:
As I said, error messages should not be considered part of the programmatic interface. They are for humans. To behave differently depending ob specific kinds of errors, error codes or classes should be used. Otherwise one day they change the wording in a minor/patch version and even though semantically the same meaning our behaviour here breaks.

My feeling here is that we don't need different behaviour depending on error types.